pre-check/v1.1.x/check.sh

```diff
@@ -305,6 +305,34 @@ check_free_space()
     record_fail
 }
 
+check_certificate()
+{
+    echo ">>> Check certificates..."
+
+    vip=$(kubectl -n kube-system get services ingress-expose -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    if [ -z "$vip" ]; then
+        echo "VIP not available."
+        record_fail
+    else
+        echo "VIP address: $vip"
+    fi
+
+    cert=$(kubectl get settings.harvesterhci.io ssl-certificates -o jsonpath='{.value}' | jq -r '.publicCertificate')
+    if [ -z "$cert" ]; then
+        echo "ssl-certificates is not set."
+    else
+        echo "Checking if the user-provided certificate contains any Subject Alternative Names (SANs)..."
+        SAN=$(echo "$cert" | openssl x509 -inform PEM -noout -ext subjectAltName | grep -oP '(?<=DNS:|IP Address:)[^,]+' || true)
+        if echo "$SAN" | grep -q "$vip"; then
+            echo "The certificate contains the VIP address $vip"
+        else
+            echo "The SAN extension of the certificate does not contain the VIP address $vip"
+            echo "For more information, please visit: https://github.com/harvester/harvester/issues/4519"
+            record_fail
```

Comment:
question: do not allow upgrade in such case? how can user fix it?

Reply:
This script only helps to check for upgrades and does not prevent any upgrade process; however, it is not recommended to proceed with the upgrade operation if it contains an error message. And I think It would be more useful if we could link to a suggested workaround once it was available.

Reply:
sounds good, let's wait for the workaround and update here again